cpp/src/replace/replace.cu

```diff
@@ -247,7 +409,91 @@ struct replace_kernel_forwarder {
                                            cudf::column_view const& replacement_values,
                                            rmm::mr::device_memory_resource* mr,
                                            cudaStream_t stream = 0) {
-    CUDF_FAIL("Non fixed-width types are not supported.");
+    rmm::device_scalar<cudf::size_type> valid_counter(0);
```

Comment:
Above, there are stream semantics issues just like discussed below in the parts of this code not touched by this PR. Please check and fix as guided below (moving streamSynchronize, using cudaMemcpyAsync/cudaMemsetAsync, stream for `device_scalar`, etc.).

Reply:
I think I've found and fixed all these issues.